src/Framework/IBuildEngineInternal.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    ///
+    /// </summary>
+    internal interface IBuildEngineInternal
```

Comment:
Did we decide we wanted to go forward with this internal IBuildEngine? I'm in favor, but I wasn't clear there was a definitive agreement.

Reply:
I've removed it and introduced the ultimate `IBuildEngine10` which exposes a new versioning mechanism.